docs/rfcs/9999-vpn-firefox-integration.md

```diff
@@ -0,0 +1,164 @@
+The following parts expect knowledge on how the MAC Extension works, for a refresher scroll to appendix :) 
+
+----
+
+## Preconditions
+
+For this feature to be enabled, a Firefox user on the desktop needs to have the VPN installed, be logged in and have an active subscription. 
+
+
+## VPN API 
+
+
+### Networking
+Using the [proxy api](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/proxy) a network extension can decide which proxy each network request should use. So from a network request perspective, there are only 4 possible cases to implement:
+
+| StateNr               | Device VPN is On | Device VPN is OFF |
+|-----------------|------------------|-------------------|
+| Page VPN is On  | 1                | 2                 |
+| Page VPN is Off | 3                | 4                 |
+
+In the following sections, we will explain what changes we need inside the client to support each of the cases and where the extension must proxy traffic to. 
+
+
+#### State 1: Device VPN is On & Page VPN is On
+In this State the Extension can simply set the Proxy for those network requests to `null` to use the user's active server.
+If a custom exit location is set it must set it to the `socks5` proxy of that endpoint. No changes are needed on the Client as this is how MAC uses the current API. 
+
+
+#### State 2: Device VPN is OFF & Page VPN is On
+
+Simplified, the VPN has the following states:
+```cpp
+  enum State {
+    StateOff, // Noting protected
+    StateOn, // Whole device protected
+    (...) // unrelated intermediate states
+  };
+```
+An extension can send `{"t":"activate"}` to the VPN to turn on whole-device protection. 
+For this case, we propose to introduce a new state `StateOnPartial` that will be activated on a call of an extension to `{"t":"activate"}` instead. It is similar to `StateOn` with the following properties: 
+- The `allowedIp` for this activation is set to `10.124.0.0/20` so that only the Mullvad Exit Proxies are accessible on the VPN Network Device. Therefore, all other traffic will not use the VPN. 
+- A Mullvad DNS server will be set, so that the Mullvad proxies are resolvable. If a custom DNS is set, we need to set a Mullvad DNS as 2nd DNS server.
+- The Client UI does not show an activation. (or a new special UI state)
+> Q: Do we need a new UI state ? or like a "firefox is using the vpn?" 
+- If the Client activates, it will attempt to switch to `StateOn` and protect the whole device.
+- An Extension may request `{"t":"deactivate"}` only if the client is in `StateOnPartial` and it was the activator. The Client will then go to `StateOff`.
+
+The extension must then set the proxy to the `socks5` URL of the current `exitServer` OR, if a custom exit location is set, to the `socks5` proxy of that endpoint.
+
+#### State 3: Device VPN is On & Page VPN is Off

```

Comment:
Do we really want to cover this scenario? It seems that we are offering a way to workaround the VPN protection. 

In theory this proxy could be abused by malicious app.

Reply:
>In theory this proxy could be abused by malicious app. 

I agree, however if we can only ship one of those states, the per tab split tunnel would be what Product would request 😅